src/api/test/activexml_matcher.rb

```diff
@@ -175,7 +175,10 @@ def self.match(node, conditions)
     if conditions[:ancestor]
       return false unless catch :found do
         p = node
-        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
+        while p
+          p = p.parent
```

Comment:
Yes. @marcus-h commented it here https://github.com/openSUSE/open-build-service/pull/1132#issuecomment-140434914
The first value of *p* should be an ancestor, not the node itself... isn't it?